lib/cli/interactive-setup/service.js

```diff
@@ -55,27 +58,78 @@ const projectNameInput = async (workingDir) =>
     })
   ).projectName.trim();
 
+const resolveProjectNameInput = async (options, workingDir) => {
+  if (options.name) {
+    if (!isValidServiceName(options.name)) {
+      throw new ServerlessError(INVALID_PROJECT_NAME_MESSAGE, 'INVALID_PROJECT_NAME');
+    }
+
+    let alreadyTaken = false;
+    try {
+      await fs.promises.access(join(workingDir, options.name));
+      alreadyTaken = true;
+    } catch {
+      // Pass
+    }
+
+    if (alreadyTaken) {
+      throw new ServerlessError(
+        `Path ${options.name} is already taken`,
+        'TARGET_FOLDER_ALREADY_EXISTS'
+      );
+    }
+
+    return options.name;
+  }
+
+  return projectNameInput(workingDir);
+};
+
 module.exports = {
-  isApplicable({ serviceDir }) {
+  isApplicable({ options, serviceDir }) {
+    if (serviceDir && (options.name || options['template-path'])) {
+      throw new ServerlessError(
+        'Project already exists in current directory. If you want to continue its setup, run "serverless" without `--name`, `--template-path` options. Otherwise, if you want to setup a new project, run the command in a non-service directory.',
+        'INTERACTIVE_SERVICE_ALREADY_EXISTS'
```

Comment:
We do not do `INTERACTIVE_` prefix in case of other errors here, so I would probably not set it here.

Maybe let's name it simply `NOT_APPLICABLE_SERVICE_OPTIONS` (?) (when processing errors we have information of command being used, so that doesn't have to leak to error codes)